src/interface/components/VLIW/modal/VLIWLoadModalComponent.tsx

```diff
@@ -61,56 +58,62 @@ export const VLIWLoadModalComponent = ({
   isLoadModalOpen,
   actions,
 }: VLIWLoadModalComponentProps) => {
-  const [modalError, setModalError] = useState("");
-  const [modalVLIWError, setModalVLIWError] = useState("");
-  const [modalCode, setModalCode] = useState(DEFAULT_MODAL_CODE);
-  const [modalVLIWCode, setModalVLIWCode] = useState(DEFAULT_MODAL_VLIW_CODE);
+  const [modalError, setModalError] = useState({ general: "", vliw: "" });
+  const [modalCode, setModalCode] = useState({
+    general: DEFAULT_MODAL_CODE,
+    vliw: DEFAULT_MODAL_VLIW_CODE,
+  });
   const [t] = useTranslation();
 
   const close = () => {
     actions.toggleLoadModal(false);
   };
 
   const loadCodeFromFile = ([[fileContent]]) => {
-    setModalCode(fileContent.target.result);
+    setModalCode({ ...modalCode, general: fileContent.target.result });
   };
 
   const loadVLIWCodeFromFile = ([[fileContent]]) => {
-    setModalVLIWCode(fileContent.target.result);
+    setModalCode({ ...modalCode, vliw: fileContent.target.result });
   };
 
   const loadCode = () => {
     const code = new Code();
     const vliwCode = new VLIWCode();
 
+    let generalError = modalError.general;
+    let vliwError = modalError.vliw;
+
     try {
-      code.load(modalCode);
-      setModalError("");
+      code.load(modalCode.general);
+      generalError = "";
```

Comment:
```suggestion
      setModalError({...modalError, general: ""});
```

Reply:
This does not work, setModalError updates Modal Error asynchronously, so you can call it once in a function call, or it will replace the data with the old one. For example, is there were a previous general error, so we call setModalError to set it to "". But later if there is a vliw error, when we set it, we read the old value of general error, not "".